File: app/src/main/java/xyz/ummo/user/data/dao/ProfileDao.java
package xyz.ummo.user.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import xyz.ummo.user.data.entity.ProductEntity;
import xyz.ummo.user.data.entity.ProfileEntity;

@Dao
public interface ProfileDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertProfile(ProfileEntity profileEntity);

    @Query("SELECT * FROM profile")
    LiveData<ProfileEntity> getProfileLiveData();

    @Query("SELECT * FROM profile")
    LiveData<ProfileEntity> getProfileEntityLiveDataById();

    @Update
    void updateProfile(ProfileEntity profileEntity);

    @Query("DELETE FROM profile")
    void deleteProfile();
}

File: app/src/main/java/xyz/ummo/user/data/dao/ProfileDao.kt
package xyz.ummo.user.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import xyz.ummo.user.data.entity.ProfileEntity

@Dao
interface ProfileDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertProfile(profileEntity: ProfileEntity?)

    @get:Query("SELECT * FROM profile")
    val profileLiveData: LiveData<ProfileEntity?>?

    @get:Query("SELECT * FROM profile")
    val profileListData: List<ProfileEntity?>?

    @get:Query("SELECT * FROM profile")
    val profileEntityLiveDataById: LiveData<ProfileEntity?>?

    @Update
    fun updateProfile(profileEntity: ProfileEntity?)

    @Query("DELETE FROM profile")
    fun deleteProfile()
}